bucket_map/src/bucket_map.rs

```diff
@@ -181,6 +182,62 @@ fn read_be_u64(input: &[u8]) -> u64 {
     u64::from_be_bytes(input[0..std::mem::size_of::<u64>()].try_into().unwrap())
 }
 
+/// Utility function to get number of Mmap files for current process
+pub fn get_mmap_count() -> Option<usize> {
+    use std::io::BufRead;
+
+    let pid = std::process::id();
+    let map_path = format!("/proc/{}/maps", pid);
+    let tmp_dir = tempfile::TempDir::new().ok()?;
+    let copy_path = tmp_dir.path().join("maps");
+    std::fs::copy(map_path, copy_path.as_os_str()).unwrap();
+
+    let file = std::fs::File::open(copy_path).ok()?;
+    Some(std::io::BufReader::new(file).lines().count())
```

Comment:
how's timing in comparison?

Reply:
```
wc-l        1.5s
wc-l-copy  1.7s
readline    3.1s
```

Reply:
looks like wc-l without copy is the fastest.

Reply:
that's unexpected :thinking: 

did you try `.lines()` w/o copy?

Reply:
yeah. .lines() w/o copy takes 2.87s
seems that `wc` is better optimized than rust lines() fn.

Reply:
weird.  i wonder if it's the utf-8 handling on the rust side.  `wc` is probably dumb and assumes ascii :thinking: